modules/python/python-main.c

```diff
@@ -305,14 +305,13 @@ _get_venv_path(void)
       return env_virtual_env;
     }
 
-
   const gchar *private_venv_path = get_installation_path_for(SYSLOG_NG_PYTHON_VENV_DIR);
   if (!_py_is_virtualenv_valid(private_venv_path))
     {
-      msg_warning("python: private virtualenv is not initialized, use the `syslog-ng-update-virtualenv' "
-                  "script to initialize it or make sure all required Python dependencies are available in "
-                  "the system Python installation",
-                  evt_tag_str("path", private_venv_path));
+      msg_debug("python: private virtualenv is not initialized, use the `syslog-ng-update-virtualenv' "
```

Comment:
I don't understand this change (why we need to move this to the debug level).

If our virtualenv is optional, that should be reflected in the value of the `use_virtualenv` flag in my opinion.

None of the virtualenv-related error messages are shown when `use_virtualenv` is false, which currently requires the user (or the syslog-ng package maintainer) to set `@module mod-python use-virtualenv(no)`.

If setting `@module mod-python use-virtualenv(no)` is problematic, we could provide another way to disable the virtualenv (compile time flag, or something that can be disabled during packaging). My understanding is that the user (or package maintainer) decides if they want to use system Python dependencies or the venv, and once that decision is made, syslog-ng should provide error messages if something isn't right.

Maybe I'm wrong and we just want to fall back silently, but in that case, the `use_virtualenv` flag seems to be unnecessary.

Reply:
What do you think about the current solution? I just pushed. Moved all messages to debug at the same time every time we get a ModuleNotFoundError we report that the user might need to create the virtualenv.

Cheers,
Bazsi